besskge/scoring.py

```diff
@@ -1175,3 +1175,150 @@ def score_tails(
             width_ht.view(-1, 1, 2, self.embedding_size),
             box_size.view(-1, 1, 2),
         )
+
+
+class InterHT(DistanceBasedScoreFunction):
+    """
+    InterHT scoring function :cite:p:`InterHT`.
+    """
+
+    def __init__(
+        self,
+        negative_sample_sharing: bool,
+        scoring_norm: int,
+        sharding: Sharding,
+        n_relation_type: int,
+        embedding_size: int,
+        entity_initializer: Union[torch.Tensor, List[Callable[..., torch.Tensor]]] = [
+            init_KGE_uniform
+        ],
+        relation_initializer: Union[torch.Tensor, List[Callable[..., torch.Tensor]]] = [
+            init_KGE_uniform
+        ],
+        normalize_entities: bool = True,
+        offset: float = 1.0,
+    ) -> None:
+        """
+        Initialize InterHT model.
+
+        :param negative_sample_sharing:
+            see :meth:`DistanceBasedScoreFunction.__init__`
+        :param scoring_norm:
+            see :meth:`DistanceBasedScoreFunction.__init__`
+        :param sharding:
+            Entity sharding.
+        :param n_relation_type:
+            Number of relation types in the knowledge graph.
+        :param embedding_size:
+            Size of entity and relation embeddings.
+        :param entity_initializer:
+            Initialization function or table for entity embeddings.
+        :param relation_initializer:
+            Initialization function or table for relation embeddings.
+        :param normalize_entities:
+            If True, L2-normalize head and tail entity embeddings before projecting.
+            Default: True.
+        :param offset:
+            Factor applied to offset for auxiliary entity embeddings. Default: 1.0.
+        """
+        super(InterHT, self).__init__(
+            negative_sample_sharing=negative_sample_sharing, scoring_norm=scoring_norm
+        )
+
+        self.sharding = sharding
+        self.normalize = normalize_entities
+
+        # self.entity_embedding[..., :embedding_size] : entity embedding
+        # self.entity_embedding[..., embedding_size:] : auxiliary entity embedding
+        self.entity_embedding = initialize_entity_embedding(
+            self.sharding, entity_initializer, [2 * embedding_size]
```

Comment:
Similarly to BoxE/PairRE/TripleRE, I would maybe allow for independent initialization of the two embedding blocks for entities (regular and auxiliary), namely have
`self.entity_embedding = initialize_entity_embedding(self.sharding, entity_initializer, [embedding_size, embedding_size])`
adding a 
```
if isinstance(entity_initializer, list):
   entity_initializer= 2 * entity_initializer
```
before.